sources/nvidia-migmanager/src/main.rs

```diff
@@ -0,0 +1,673 @@
+const NVIDIA_SMI_PATH: &str = "/usr/libexec/nvidia/tesla/bin/nvidia-smi";
+const TOUCH_PATH: &str = "usr/bin/touch";
+
+/// Stores user-supplied global arguments
+struct Args {
+    log_level: LevelFilter,
+    config_path: PathBuf,
+}
+
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            log_level: LevelFilter::Info,
+            config_path: PathBuf::from_str(DEFAULT_CONFIG_PATH).unwrap(),
+        }
+    }
+}
+
+/// Stores the subcommand to be executed
+#[derive(Debug)]
+enum Subcommand {
+    HandleMIGManager,
+    StopFabricManager,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct NvidiaMIGConfig {
+    #[serde(default)]
+    device_partitioning_strategy: NvidiaDevicePartitioningStrategy,
+    #[serde(default)]
+    profile: HashMap<NvidiaGPUModel, MIGProfile>,
+}
+
+#[derive(Debug, PartialEq)]
+enum NvidiaGPU {
+    A100_40GB,
+    A100_80GB,
+    H100_80GB,
+    H200_141GB,
+    Other,
+}
+
+#[derive(Hash, Debug, Clone, PartialEq, Eq)]
+enum MigState {
+    Unsupported,
+    Enabled,
+    Disabled,
+    Transition,
+    Unknown,
+}
+
+/// Wrapper around process::Command that adds error checking.
+fn command<I, S>(bin_path: &str, args: I) -> Result<String>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let mut command = Command::new(bin_path);
+    command.args(args);
+    let output = command
+        .output()
+        .context(error::ExecutionFailureSnafu { command })?;
+
+    trace!("stdout: {}", String::from_utf8_lossy(&output.stdout));
+    trace!("stderr: {}", String::from_utf8_lossy(&output.stderr));
+
+    ensure!(
+        output.status.success(),
+        error::CommandFailureSnafu { bin_path, output }
+    );
+
+    let output_str = String::from_utf8_lossy(&output.stdout);
+
+    Ok(output_str.to_string())
+}
+
+// Runs the nvidia-smi command to enable/disable MIG in all GPUs
+fn set_mig_mode(mig_enabled: bool) -> Result<()> {
+    info!(
+        "{} MIG.",
+        if mig_enabled { "Enabling" } else { "Disabling" }
+    );
+
+    command(NVIDIA_SMI_PATH, ["-mig", &(mig_enabled as u8).to_string()])?;
+    Ok(())
+}
+
+// Runs the nvidia-smi command to apply the correct MIG profile in all the GPUs
+fn set_mig_profile(profile_string: &str) -> Result<()> {
+    info!("Activating MIG profile.");
+
+    command(NVIDIA_SMI_PATH, ["mig", "-cgi", profile_string, "-C"])?;
+    Ok(())
+}
+
+// Runs nvidia-smi command to find out the current state and the next state.
+fn get_gpu_state() -> Result<MigState> {
+    let output = command(
+        NVIDIA_SMI_PATH,
+        [
+            "--query-gpu=gpu_name,mig.mode.current,mig.mode.pending",
+            "--format=csv,noheader",
+        ],
+    )?;
+
+    let mut current_states = HashSet::new();
+    let mut pending_states = HashSet::new();
+    let mut gpu_state = MigState::Unknown;
+
+    for line in output.lines() {
+        let parts: Vec<_> = line.split(", ").collect();
```

Comment:
Does the CSV output format really have whitespace after the commas?
```suggestion
        let parts: Vec<_> = line.split(",").collect();
```

Reply:
Yes. I ran the command in g4dn.xlarge and got this output

```
Tesla T4, [N/A], [N/A]
```